api-definitions/common/job-status.yaml

```diff
@@ -0,0 +1,29 @@
+################################################################################
+# This schema defines the common elements of a Job Status response. Individual
+# job types may extend this model with task-specific properties, but these
+# common properties must be present in the response to all job status queries.
+################################################################################
+
+schemas:
+  JobStatus:
+    type: object
+    description: The status of a job in the system
+    properties:
+      job_id:
+        type: string
+        description: Unique identifier for a single job
+      status:
+        type: string
+        enum:
+          - QUEUED
+          - RUNNING
+          - COMPLETED
+          - CANCELED
+          - ERRORED
```

Comment:
Should we map with whatever Kubernetes says on Job'status?

Reply:
Do we feel we need to model anything for when a job goes through a "temporary failure" and let's say goes through a retry? Or would we jusft go from FAILED to QUEUED again? Or would we just consider that "process" as another job entirely. 

Just thinking through how we would like to look at modeling what could happen let's say when a job hits a transient failure (let's say due to part of it running on bad infrastructure that is then replaced) and then a retry of that is scheduled.

Reply:
I like the enum too as well as the remap from the Kube terminology. Thanks!